persistence/trees/module_test.go

```diff
@@ -0,0 +1,120 @@
+	// TODO: Write test case for the DebugClearAll method
+	t.Skip("TODO: Write test case for DebugClearAll method")
+}
+
+// createMockGenesisState configures and returns a mocked GenesisState
+func createMockGenesisState(valKeys []cryptoPocket.PrivateKey) *genesis.GenesisState {
+	genesisState := new(genesis.GenesisState)
+	validators := make([]*coreTypes.Actor, len(valKeys))
+	for i, valKey := range valKeys {
+		addr := valKey.Address().String()
+		mockActor := &coreTypes.Actor{
+			ActorType:       coreTypes.ActorType_ACTOR_TYPE_VAL,
+			Address:         addr,
+			PublicKey:       valKey.PublicKey().String(),
+			ServiceUrl:      validatorId(i + 1),
+			StakedAmount:    test_artifacts.DefaultStakeAmountString,
+			PausedHeight:    int64(0),
+			UnstakingHeight: int64(0),
+			Output:          addr,
+		}
+		validators[i] = mockActor
+	}
+	genesisState.Validators = validators
+
+	return genesisState
+}
+
+// Persistence mock - only needed for validatorMap access
+func preparePersistenceMock(t *testing.T, busMock *mockModules.MockBus, genesisState *genesis.GenesisState) *mockModules.MockPersistenceModule {
+	ctrl := gomock.NewController(t)
+
+	persistenceModuleMock := mockModules.NewMockPersistenceModule(ctrl)
+	readCtxMock := mockModules.NewMockPersistenceReadContext(ctrl)
+
+	readCtxMock.EXPECT().GetAllValidators(gomock.Any()).Return(genesisState.GetValidators(), nil).AnyTimes()
+	readCtxMock.EXPECT().GetAllStakedActors(gomock.Any()).DoAndReturn(func(height int64) ([]*coreTypes.Actor, error) {
+		return testutil.Concatenate[*coreTypes.Actor](
+			genesisState.GetValidators(),
+			genesisState.GetServicers(),
+			genesisState.GetFishermen(),
+			genesisState.GetApplications(),
+		), nil
+	}).AnyTimes()
+	persistenceModuleMock.EXPECT().NewReadContext(gomock.Any()).Return(readCtxMock, nil).AnyTimes()
+	readCtxMock.EXPECT().Release().AnyTimes()
+
+	persistenceModuleMock.EXPECT().GetBus().Return(busMock).AnyTimes()
+	persistenceModuleMock.EXPECT().SetBus(busMock).AnyTimes()
+	persistenceModuleMock.EXPECT().GetModuleName().Return(modules.PersistenceModuleName).AnyTimes()
+	busMock.RegisterModule(persistenceModuleMock)
+
+	return persistenceModuleMock
+}
+
+func validatorId(i int) string {
+	return fmt.Sprintf(serviceURLFormat, i)
+}
+
+// createMockBus returns a mock bus with stubbed out functions for bus registration
+func createMockBus(t *testing.T, runtimeMgr modules.RuntimeMgr) *mockModules.MockBus {
+	t.Helper()
+	ctrl := gomock.NewController(t)
+	mockBus := mockModules.NewMockBus(ctrl)
+	mockBus.EXPECT().GetRuntimeMgr().Return(runtimeMgr).AnyTimes()
+	mockBus.EXPECT().RegisterModule(gomock.Any()).DoAndReturn(func(m modules.Submodule) {
+		m.SetBus(mockBus)
+	}).AnyTimes()
+	mockModulesRegistry := mockModules.NewMockModulesRegistry(ctrl)
+	mockModulesRegistry.EXPECT().GetModule(peerstore_provider.PeerstoreProviderSubmoduleName).Return(nil, runtime.ErrModuleNotRegistered(peerstore_provider.PeerstoreProviderSubmoduleName)).AnyTimes()
```

Comment:
NIT: I think multilining these more complex mocks really makes it much easier to understand what's being called & returned.

Ditto elsewhere.

```suggestion
mockModulesRegistry.EXPECT().
	GetModule(peerstore_provider.PeerstoreProviderSubmoduleName).
	Return(nil, runtime.ErrModuleNotRegistered(peerstore_provider.PeerstoreProviderSubmoduleName)).
	AnyTimes()
```